source/agora/test/PeriodicCatchup.d

```diff
@@ -63,6 +59,12 @@ private extern(C++) class DoesNotExternalizeBlockNominator : TestNominator
             assert(0, format!"PeriodicCatchup exception thrown during test: %s"(e));
         }
     }
+
+    extern(D) public override void receiveBlockSignature (in ValidatorBlockSig block_sig) @safe
+    {
+        if (block_sig.height == 1)
+            log.trace("Ignore signatures for block 1 to test signature catchup");
+    }
```

Comment:
Why this change ?

Reply:
This adds a test for missing signatures to the `PeriodicCatchup` Network test. I'll extract it to a separate commit with the trace log line above (which triggered me to update this test).

Reply:
In fact this was a required change as the validators can only be added (once a bit is set it can not be unset).